### PR TITLE
[Task Submission] Natural Language Codesearch Classification (`nl_codesearch_clf `)

### PR DESCRIPTION
**Pull request by drndr**
# [Natural Language Codesearch Classification]
The task consists of 8 subtasks and measures cross-lingual and domain generalization, and robustness to covariate shift
Includes a binary classification evaluation:
Given a natural language query, determine if a given code snippet is matches the natural language description or not.
## Authors
- Andor Diera `andor.diera@uni-ulm.de`
- Abdelhalim Dahou `Abdelhalim.Dahou@gesis.org`
- Lukas Galke `Lukas@LPAG.de`
- Fabian Karl `fabian.karl@uni-ulm.de`
- Florian Sihler `florian.sihler@uni-ulm.de`
- Ansgar Scherp `ansgar.scherp@uni-ulm.de`

## Implementation
For the binary classification setup the config files were used with task_type "multiple_choice"
The task.py script includes a custom get_dataset_raw method, where the negativ samples are created.

## Usage

For binary classification the default "multiple_choice" usage.

## Checklist:

- [x] I and my co-authors agree that, if this PR is merged, the code will be available under the [same license](LICENSE) as the genbench_cbt repository.
- [x] Prior to submitting, I have ran the GenBench CBT test suite using the `genbench-cli test-task` tool.
- [x] I have read the description of what should be in the doc.md of my task, and have added the required arguments.
- [x] I have submitted or will submit an accompanying paper to the [GenBench workshop](https://genbench.org/workshop/).

### CODE REVIEW

**Comment by drndr** [comment]
test_task failed due to non existent id
our submission includes two tasks nl_codesearch_clf and nl_codesearch_mrr

**Comment by vernadankers** [comment]
Hi, that sounds like something that can be fixed on our side! Could you open an issue for that, please? :-)

**Comment by drndr** [comment]
Thanks! i was also considering submitting two different branches, but if it can be fixed on your side even better :)

**Comment by kazemnejad** [comment]
Hi, the problem was that the automated CI was trying to read the task_id (which can be the parent task, but, your initial title contained `nl_codesearch`. `nl_codesearch` didn't exist as a task_id. I changed the title to contain `nl_codesearch_clf` and it passed all the test cases. 

You are submitting two parent tasks `nl_codesearch_clf` and `nl_codesearch_mrr`. Currently, we only support one per task-submission. Please, open another PR for the second task.

**Comment by drndr** [comment]
Thanks! I removed the mrr part from this clf PR, and created a seperate one for mrr

**Comment by vernadankers** [comment]
Hello!

We are getting quite close to the deadline (September 1, 11:59PM anywhere on earth), so if your PR needs any final changes, please make them now,
and don't forget to submit your accompanying paper to Openreview via https://openreview.net/group?id=GenBench.org/2023/Workshop by September 1.

Good luck finalising your PR and paper, feel free to tag us if you have questions.
Cheers, Verna
On behalf of the GenBench team

**Comment by drndr** [comment]
Hi,

Sorry there were some style issues introduced in the last commit at submission and forgot the rerun the make commands.
Fixed these issues in the latest commit.

**Comment by kazemnejad** [comment]
@drndr We're in the process of merging the tasks into the repo.

Could you please include a single file usage_example.py of each task where you use each task. It seems your tasks supports both finetuning/in-context-learning. It'd be nice to have examples of both usages. (Preferably a pretrained huggingface model). 

Please also include requirements-usage-example.txt for the python dependencies needed to be installed for running the example.

File: src/genbench/tasks/nl_codesearch_clf/usage_example.py [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+    parser.add_argument("--output_dir", type=str, default="models")
+    parser.add_argument("--seq_len", type=int, default=512, help="maximum sequence length")
+    # parser.add_argument("--distractors", type=int, default=99, help="number of distractors per true pair")
+    parser.add_argument("--log_level", choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"], default="INFO")
+
+    args = parser.parse_args()
+
+    # TRAIN_FILE = "./codesearchnet_adv/train_adv_clf.jsonl"
+
+    TRAIN_FILE = NlCodesearchClfCodesearchnetAdv(
```
Comment (kazemnejad) [inline src/genbench/tasks/nl_codesearch_clf/usage_example.py]: This looks great. Just a minor comment. Could you replace the task loading using the official method? for here and the rest of this file?

```python
from genbench import load_task
task = load_task("nl_codesearch_clf:codesearchnet_adv")
```

**Comment by kazemnejad** [comment]
Manually merged in 
https://github.com/GenBench/genbench_cbt/pull/41

**Pull request closed without merging**